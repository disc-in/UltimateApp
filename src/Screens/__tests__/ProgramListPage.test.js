import React from 'react';
import { Provider } from 'react-redux';
import { render, fireEvent, waitFor } from '@testing-library/react-native';

import store from '../../Store/testStore';
import { createProgram, createTraining } from '../../Fixtures/TestFixtures';
import { DrillTypes, EquipmentLabels } from '../../Fixtures/config';

import ConnectedProgramListPage, { ProgramListPage } from '../ProgramListPage';

describe('<ProgramListPage />', () => {
  const route = {
    params: {
      type: DrillTypes.FITNESS,
      equipmentLabel: EquipmentLabels.NONE,
    },
  };
  it('renders correctly when connected', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <ConnectedProgramListPage route={route} />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('links to training', async () => {
    const navigation = { navigate: jest.fn() };

    const program1 = createProgram({ title: 'Program 1' });
    const firstTraining = createTraining({ id: 1, title: 'First Training' });
    const secondTraining = createTraining({ id: 2, title: 'Second Training' });
    const program2 = createProgram({ title: 'Program 2', trainings: [firstTraining, secondTraining] });
    const programs = [program1, program2];

    const { getByText } = render(
      <Provider store={store}>
        <ProgramListPage route={route} programs={programs} navigation={navigation} completeTrainings={[]} />
      </Provider>,
    );

    expect(getByText('Program 1')).toBeDefined();
    expect(getByText('Program 2')).toBeDefined();

    fireEvent.press(getByText('Program 2'));

    expect(getByText('1 - First Training')).toBeDefined();
    expect(getByText('2 - Second Training')).toBeDefined();

    fireEvent.press(getByText('1 - First Training'));

    await waitFor(() => {
      expect(navigation.navigate).toBeCalledWith('TrainingPage', { program: program2, training: firstTraining });
    });
  });
});
