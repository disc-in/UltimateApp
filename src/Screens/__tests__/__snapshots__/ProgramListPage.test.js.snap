// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgramListPage /> renders correctly when connected 1`] = `
<RCTScrollView
  style={
    {
      "height": "100%",
    }
  }
>
  <View>
    <View
      style={
        {
          "marginBottom": 10,
        }
      }
    >
      <View
        style={{}}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": false,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#fff",
                "elevation": 3,
                "marginTop": 10,
                "padding": 10,
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 3,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "backgroundColor": "#fff",
                  "minHeight": 40,
                  "paddingBottom": 10,
                  "paddingHorizontal": 20,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#232b2b",
                      "fontSize": 20,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Beginner
                </Text>
                <Text
                  style={
                    {
                      "color": "#232b2b",
                      "fontSize": 14,
                      "textAlign": "right",
                    }
                  }
                >
                  0/2
                </Text>
              </View>
              <View
                style={
                  {
                    "backgroundColor": "#f1f1f1",
                    "borderRadius": 2,
                    "height": 3,
                    "marginTop": 5,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      {
                        "backgroundColor": "#2a9e91",
                        "borderRadius": 5,
                        "width": "50%",
                      },
                      {
                        "width": "0%",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "height": 0,
              "overflow": "hidden",
            }
          }
        >
          <View
            collapsable={false}
            forwardedRef={[Function]}
            onLayout={[Function]}
            style={{}}
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomColor": "#e5e5e5",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "padding": 10,
                  "paddingLeft": 30,
                }
              }
            >
              <View
                style={
                  {
                    "flexBasis": "80%",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#232b2b",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  1
                   - 
                  Dump & Swing
                </Text>
                <Text
                  style={
                    {
                      "color": "#232b2b",
                      "fontSize": 14,
                    }
                  }
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec scelerisque dignissim felis, at tristique mi efficitur a. Nulla pellentesque odio nunc, ut sodales ex pulvinar at. In hac habitasse platea dictumst. Pellentesque sodales nisl lorem, ac lacinia nisl lacinia eu. Proin lacinia viverra mauris, et pharetra ipsum tempus eget. Ut maximus sapien in hendrerit ultricies. Maecenas vel diam tincidunt, consequat nunc a, mattis eros. Curabitur a eros a nulla malesuada gravida. Donec massa quam, suscipit eget efficitur et, euismod sed nulla. Morbi mattis et magna a aliquam. Fusce pellentesque vel erat eget
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexBasis": "20%",
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomColor": "#e5e5e5",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "padding": 10,
                  "paddingLeft": 30,
                }
              }
            >
              <View
                style={
                  {
                    "flexBasis": "80%",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#232b2b",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2
                   - 
                  Training 2
                </Text>
                <Text
                  style={
                    {
                      "color": "#232b2b",
                      "fontSize": 14,
                    }
                  }
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec scelerisque dignissim felis, at tristique mi efficitur a. Nulla pellentesque odio nunc, ut sodales ex pulvinar at. In hac habitasse platea dictumst. Pellentesque sodales nisl lorem, ac lacinia nisl lacinia eu. Proin lacinia viverra mauris, et pharetra ipsum tempus eget. Ut maximus sapien in hendrerit ultricies. Maecenas vel diam tincidunt, consequat nunc a, mattis eros. Curabitur a eros a nulla malesuada gravida. Donec massa quam, suscipit eget efficitur et, euismod sed nulla. Morbi mattis et magna a aliquam. Fusce pellentesque vel erat eget
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexBasis": "20%",
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
